fix: helm install karpenter failed if ns not created

**What happened**:
```sh
➜  karpenter-provider-alicloud git:(main) make apply
2024/11/07 21:20:24 Using base cgr.dev/chainguard/static:latest@sha256:1c785f2145250a80d2d71d2b026276f3358ef3543448500c72206d37ec4ece37 for github.com/cloudpilot-ai/karpenter-provider-alibabacloud/cmd/controller
2024/11/07 21:20:24 git doesn't contain any tags. Tag info will not be available
2024/11/07 21:20:24 Building github.com/cloudpilot-ai/karpenter-provider-alibabacloud/cmd/controller for linux/arm64
2024/11/07 21:20:24 git doesn't contain any tags. Tag info will not be available
2024/11/07 21:20:24 Building github.com/cloudpilot-ai/karpenter-provider-alibabacloud/cmd/controller for linux/amd64
2024/11/07 21:20:33 Loading ko.local:e95a9413b513d373523b8ea01a649e3450180267e2fa4e9cb7a26806b17c3d34
2024/11/07 21:20:35 Loaded ko.local:e95a9413b513d373523b8ea01a649e3450180267e2fa4e9cb7a26806b17c3d34
2024/11/07 21:20:35 Adding tag latest
2024/11/07 21:20:36 Added tag latest
helm upgrade --install karpenter charts/karpenter --namespace karpenter-system \
         \
        --set logLevel=debug \
        --set controller.image.repository=ko.local \
        --set controller.image.tag=e95a9413b513d373523b8ea01a649e3450180267e2fa4e9cb7a26806b17c3d34 \

Release "karpenter" does not exist. Installing it now.
Error: create: failed to create: namespaces "karpenter-system" not found
make: *** [Makefile:55: apply] Error 1
```
**What you expected to happen**:
Install karpenter successfully even if ns doesn't be created beforehand

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- Karpenter-provider-alibabacloud version (use `git describe --tags --dirty --always`):
- AliCloud ACK version:
- Others:

--- a/Makefile
+++ b/Makefile
@@ -52,11 +52,13 @@ image: ## Build the Karpenter controller images using ko build
 	$(eval IMG_TAG=$(shell echo $(CONTROLLER_IMG) | cut -d "@" -f 1 | cut -d ":" -f 2 -s))
 
 apply: image ## Deploy the controller from the current state of your git repository into your ~/.kube/config cluster
-	helm upgrade --install karpenter charts/karpenter --namespace ${KARPENTER_NAMESPACE} \
-        $(HELM_OPTS) \
-        --set logLevel=debug \
-        --set controller.image.repository=$(IMG_REPOSITORY) \
-        --set controller.image.tag=$(IMG_TAG) \
+	helm upgrade --install karpenter charts/karpenter \
+		--create-namespace \
+		--namespace ${KARPENTER_NAMESPACE} \
+		$(HELM_OPTS) \
+		--set logLevel=debug \
+		--set controller.image.repository=$(IMG_REPOSITORY) \
+		--set controller.image.tag=$(IMG_TAG) \
 
 delete: ## Delete the controller from your kubeconfig cluster
 	helm uninstall karpenter --namespace ${KARPENTER_NAMESPACE}

